### PR TITLE
[INLONG-3773][Sort-Standalone] Support configurable handler to transform data of Kafka

### LINKED ISSUE #3773: [Feature][Sort-Standalone] Support configurable handler to transform data of Kafka
### Description

Sort-Standalone support configurable handler to transform data of Kafka Sort.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luchunliang**
### Title Name: [INLONG-3773][Sort-Standalone] Sort-Standalone support a configurable handler to transform data of Kafka Sort.

Fixes #3773 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-dao/pom.xml [hunk 1]
```diff
@@ -57,8 +57,31 @@
...
+                </exclusion>
             </exclusions>
         </dependency>
+        
+        <dependency>
+            <groupId>com.sun</groupId>
+            <artifactId>tools</artifactId>
+                <version>1.8.0</version>
+                <scope>system</scope>
+                <systemPath>${JAVA_HOME}/lib/tools.jar</systemPath>
```
Comment (healchow) [inline inlong-manager/manager-dao/pom.xml:76]: Can this configuration work in Windows OS?

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsSinkContext.java [hunk 2]
```diff
@@ -17,13 +17,7 @@
...
 package org.apache.inlong.sort.standalone.sink.elasticsearch;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.LinkedBlockingQueue;
-import java.util.concurrent.atomic.AtomicLong;
+import com.alibaba.fastjson.JSON;
```
Comment (healchow) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsSinkContext.java:20]: There are too many security risks in this package, could you replace it with the Gson?

**Review by dockerzhang — APPROVED** [review]

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsSinkContext.java [hunk 3]
```diff
@@ -565,21 +550,25 @@ public void setUseIndexId(boolean isUseIndexId) {
...
+                DefaultEvent2IndexRequestHandler.class.getName());
+        try {
+            Class<?> handlerClass = ClassUtils.getClass(indexRequestHandlerClass);
+            Object handlerObject = handlerClass.getDeclaredConstructor().newInstance();
+            if (handlerObject instanceof IEvent2IndexRequestHandler) {
+                IEvent2IndexRequestHandler handler = (IEvent2IndexRequestHandler) handlerObject;
+                return handler;
+            }
+        } catch (Throwable t) {
+            LOG.error("Fail to init IEvent2IndexRequestHandler,handlerClass:{},error:{}",
```
Comment (healchow) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsSinkContext.java:569]: In the error log, it is suggested to print all exception stacks.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/kafka/DefaultEvent2KafkaRecordHandler.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+ * 
+ * DefaultEvent2KafkaRecordHandler
+ */
+public class DefaultEvent2KafkaRecordHandler implements IEvent2KafkaRecordHandler {
+
+    public static final Logger LOG = InlongLoggerFactory.getLogger(DefaultEvent2KafkaRecordHandler.class);
+
+    public static final String KEY_EXTINFO = "extinfo";
+    protected final ByteArrayOutputStream outMsg = new ByteArrayOutputStream();
+    protected final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
```
Comment (healchow) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/kafka/DefaultEvent2KafkaRecordHandler.java:41]: The `SimpleDateFormat` is not thread-safe, will there be a problem here?

**Review by EMsnap — APPROVED** [review]

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsSinkContext.java (hunk 2 above)
Comment (luchunliang) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsSinkContext.java:20]: change to com.fasterxml.jackson.databind.ObjectMapper

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsSinkContext.java (hunk 3 above)
Comment (luchunliang) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsSinkContext.java:570]: ok, print all exception stacks.

**Review by healchow — APPROVED** [review]

**Review by gosonzhang — CHANGES_REQUESTED** [review]
The postgrepsql is not used in sort-standalone, why change the main pom?

**Comment by luchunliang** [comment]
Rebase error, add new PR:https://github.com/apache/incubator-inlong/pull/3902;
This PR will close.

**Pull request closed without merging**